arbnode/batch_poster.go

```diff
@@ -1069,6 +1070,121 @@ func (s *batchSegments) SetWaitingForValidation() {
+	if !ok {
+		return nil, errors.New("failed to find add batch method")
+	}
+	// initially constructing the calldata using the old oldSequencerBatchPostMethodName method
+	// This will allow us to get the attestation quote on the hash of the data
+	args = append(args, seqNum)
+	args = append(args, l2MessageData)
+	args = append(args, new(big.Int).SetUint64(delayedMsg))
+	args = append(args, b.config().gasRefunder)
+	args = append(args, new(big.Int).SetUint64(uint64(prevMsgNum)))
+	args = append(args, new(big.Int).SetUint64(uint64(newMsgNum)))
+
+	// Later append the delay proof if needed for getting the attestion quote.
+	// If not, only append at the end of the calldata as done below.
+	calldata, err := method.Inputs.Pack(args...)
+	if err != nil {
+		return nil, err
+	}
+
+	attestationQuote, err := b.streamer.getAttestationQuote(calldata)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get attestation quote: %w", err)
+	}
+	// Construct the calldata with attestation quote
+	method, ok = b.seqInboxABI.Methods[newSequencerBatchPostMethodName]
+	if !ok {
+		return nil, errors.New("failed to find add batch method")
+	}
+	args = append(args, attestationQuote)
+
+	calldata, err = method.Inputs.Pack(args...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fullCalldata := append([]byte{}, method.ID...)
+	fullCalldata = append(fullCalldata, calldata...)
+	return fullCalldata, nil
+}
+
+func (b *BatchPoster) getCalldataForEspressoBlobBatch(
+	seqNum *big.Int,
+	prevMsgNum arbutil.MessageIndex,
+	newMsgNum arbutil.MessageIndex,
+	l2MessageData []byte,
+	delayedMsg uint64,
+) ([]byte, error) {
+	var args []any
+	method, ok := b.seqInboxABI.Methods[oldSequencerBatchPostWithBlobsMethodName]
+	if !ok {
+		return nil, errors.New("failed to find add batch method")
+	}
+	kzgBlobs, err := blobs.EncodeBlobs(l2MessageData)
+	_, blobHashes, err := blobs.ComputeCommitmentsAndHashes(kzgBlobs)
+	// initially constructing the calldata using the old SequencerBatchPostWithBlobsMethodName method
+	// This will allow us to get the attestation quote on the hash of the data
+	args = append(args, seqNum)
+	args = append(args, new(big.Int).SetUint64(delayedMsg))
+	args = append(args, b.config().gasRefunder)
+	args = append(args, new(big.Int).SetUint64(uint64(prevMsgNum)))
+	args = append(args, new(big.Int).SetUint64(uint64(newMsgNum)))
+
+	// Later append the delay proof if needed for getting the attestion quote.
+	// If not, only append at the end of the calldata as done below.
+	calldata, err := method.Inputs.Pack(args...)
+	if err != nil {
+		return nil, err
+	}
+	// bytes32Type, err := abi.NewType("bytes32", "", nil)
+	// if err != nil{
+	//   return nil, err
+	// }
+	bytes32ArrayType, err := abi.NewType("bytes32[]", "", nil)
+	if err != nil {
+		return nil, err
+	}
+	packedBlobHashes, err := abi.Arguments{
+		{Type: bytes32ArrayType},
+	}.Pack(blobHashes)
+	quoteData := append(calldata, packedBlobHashes...)
```

Comment:
Oh I see what Jeremy was saying in the morning, is this right? I dont think this is right because what contract does is 
abi.encode(seqNum, ....., abi.encode(hashes). We cant just append here

Reply:
Oh I think I see what you're saying. I assumed because abi.encode appends the dynamic length data at the end of the byte array it generates, I would be able to simply append it to the end of the call data, but I see that would result in the offset and length fields at the beginning of the calldata being wrong (if they are there at all)

Generating an `abi.Arguments` object that would take all of this data and then calling pack on that instead of `method.Inputs`  would that solve the issue correct?

Reply:
yes I think that would be the way to go